File: sdk/python/oauth.py
import json
from urllib.parse import urlparse
from urllib.parse import parse_qs
from fattureincloud_python_sdk.oauth2.oauth2 import OAuth2AuthorizationCodeManager
from fattureincloud_python_sdk.oauth2.oauth2 import Scope

class Oauth:
	def get_oauth_access_token(self):
		oauth = OAuth2AuthorizationCodeManager('CLIENT_ID', 'CLIENT_SECRET', 'http://localhost:8000/oauth')
		query_components = parse_qs(urlparse(self.path).query)
		if 'code' in query_components:
			self.send_response(200)
			self.send_header('Content-type','text/html')
			self.end_headers()
			code = query_components['code'][0]
			token = oauth.fetch_token(code)
			file = open('./token.json', 'w')
			file.write(json.dumps({"access_token": token.access_token}))  #saving the oAuth access token in the token.json file
			file.close()
			self.wfile.write(bytes('Token saved succesfully in ./token.json', 'utf8'))
		else:
			url = oauth.get_authorization_url([Scope.ENTITY_SUPPLIERS_READ], 'EXAMPLE_STATE')
			self.send_response(302)
			self.send_header('Location', url)
			self.end_headers()
		return

File: sdk/ruby/oauth.rb
require 'rubygems'
require 'webrick'
require 'json'
require 'fattureincloud_ruby_sdk'

class Oauth < WEBrick::HTTPServlet::AbstractServlet
	def do_GET(request, response)
		oauth = FattureInCloud_Ruby_Sdk::OAuth2AuthorizationCodeManager.new('CLIENT_ID', 'CLIENT_SECRET', 'http://localhost:8000/oauth')
		if !request.request_uri.query.nil?
			url_obj = URI.decode_www_form(request.request_uri.query).to_h
			if !url_obj['code'].nil?
				token = oauth.fetch_token(url_obj['code'])
				File.open('./token.json', 'w') do |file|
					file.write({"access_token" => token.access_token}.to_json) # saving the oAuth access token in the token.json file
				end
				body = 'Token saved succesfully in ./token.json'
			else
				redirect(response, oauth)
			end
		else redirect(response, oauth)
		end

		response.status = 200
		response['Content-Type'] = 'text/html'
		response.body = body
	end

	def redirect(response, oauth)
		url = oauth.get_authorization_url([FattureInCloud_Ruby_Sdk::Scope::ENTITY_SUPPLIERS_READ], 'EXAMPLE_STATE')
		response.set_redirect(WEBrick::HTTPStatus::TemporaryRedirect, url)
	end
end

if $PROGRAM_NAME == __FILE__
	server = WEBrick::HTTPServer.new(Port: 8000)
	server.mount '/oauth', Oauth
	trap 'INT' do server.shutdown end
	server.start
end
